### ISSUE TITLE
Create devel branch

### ISSUE THREAD

**Issue by mitar** [meteor/blaze#7]
So that `master` is the stable branch.

**Comment by rgoomar** [comment]
I would argue why not have just a `master` branch with tags / releases? If people are pointing to the `master` branch, they are really just pointing to the latest release. In my opinion, you only really need one main branch and develop on top of that and tag / release at specific points. 

Thoughts on that? I know the main `meteor` repository follows the `devel` and `master` model, but I don't think it is really necessary. 

In terms of using `master` for automation once a release is ready if you have a `devel` branch, you could also do that on tags that are in the specific semver format for release (i.e. `major.minor.patch` instead of `major.minor.patch-beta`).

**Comment by rgoomar** [comment]
I guess the only issue that leaves is bug fixes that need to be released outside of what is in development now like a hotfix.

**Comment by mitar** [comment]
Provably what I am advocating for: http://nvie.com/posts/a-successful-git-branching-model/

**Comment by mitar** [comment]
So what would be the integration branch before the release? It is common to name it `devel`.

**Comment by rgoomar** [comment]
So, I disagree with having to use that model. I don't see the benefits in having those separate branches. Here's the way I think works really well and [nodejs](https://github.com/nodejs/node) follows it as well. 

You have a `master` branch that is where all of your development happens. You have separate release branches, that way if there happens to be some sort of "LTS" of the product, you can keep that in a branch and if there is a critical fix that needs to be done, it can go into both that release branch and back into `master`. 

I'll leave this here - https://news.ycombinator.com/item?id=9744059

But, anyways, this argument can go on for awhile and in the end, it's up to the maintainers to decide the model that works best for them. Just thought I would put my two cents about it.

**Comment by mitar** [comment]
Hm, thank you for pointing all this out. An interesting read.

**Comment by stubailo** [comment]
Personally I think what branch is named `master` is not important - you could for example have a branch called `stable` and have `master` be unstable. Honestly I don't think it matters that much.

**Comment by mitar** [comment]
I think the point was that we should not have two branches (stable and development), but just one, where releases have their own branches, if needed to bugfix something.

**Comment by stubailo** [comment]
Ah, yes. For example right now in meteor/meteor all actual development is happening on `release-1.3` branch, even though we have `devel`. So perhaps that validates that `devel` is actually useless.

**Comment by mitar** [comment]
Yes, but the argument is that you should be working on `master` for the next release (BTW, why are you not using `devel`?). So you can use tags to know where was `1.2.1` release, you do not need whole master branch for that.

I really like this [read pointed above](http://endoflineblog.com/gitflow-considered-harmful), check it out.

Also, you should clean all those unnecessary old branches in Meteor. Why are you keeping them around?

**Comment by stubailo** [comment]
> BTW, why are you not using devel?

Just because at some point someone decided to create a release branch and we never merged it into `devel`. I guess what I am saying is that might be a more natural model since that's basically what we ended up doing by accident.

> Also, you should clean all those unnecessary old branches in Meteor. Why are you keeping them around?

Because that doesn't seem like a good use of time compared to some other things. Is someone really browsing the entire list of branches? If there are important branches they should be exposed via some other mechanism.

**Comment by mitar** [comment]
I am closing this in favor of simplicity.

I would propose we just keep everything in `master` and use tags to mark releases. If we have to patch something for old release, we go to that tag, patch it, commit there, tag again a new release, and this is it.

We still do branches for code changes and do internal pull requests as well for code review and discussion. But I would suggest we do delete branches once they have been merged into `master`.
